File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] solutions = new int[2];

        for(int i = 0; i < nums.length; i++){
            for(int j = i+1; j < nums.length; j++){
                if((nums[i] + nums[j]) == target){
                    solutions[0] = i;
                    solutions[1] = j;
                    break;
                }
            }
            if((solutions[0] + solutions[1]) == target){
                break;
            }
        }

        return solutions;
    }
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    const map = new Map();

    for(let i = 0; i < nums.length; i++){
        
        const k = target - nums[i];

        if(map.has(k)){
            return [i, map.get(k)]
        }else{
            map.set(nums[i], i)
        }
    }
};